File: scripts/checkin-daemon/test-read.js
const fs = require('fs');
const readline = require('readline');

const port = '/dev/ttyACM0';
console.log(`Abriendo puerto ${port} en modo lectura...`);

try {
  const stream = fs.createReadStream(port);
  const rl = readline.createInterface({
    input: stream,
    crlfDelay: Infinity
  });

  rl.on('line', (line) => {
    const clean = line.trim();
    if (clean) {
      console.log(`Código escaneado: [${clean}] (longitud: ${clean.length})`);
    }
  });

  stream.on('error', (err) => {
    console.error('Error en el puerto:', err.message);
    console.log('Asegúrate de tener permisos (puedes necesitar correr con sudo o estar en el grupo dialout).');
  });
} catch (e) {
  console.error('Fallo al inicializar:', e.message);
}

File: scripts/checkin-daemon/test_read.py
import serial
import sys

port = '/dev/ttyACM0'
baudrate = 9600

print(f"Abriendo {port} a {baudrate} baudios con pyserial (diagnóstico)...")
try:
    ser = serial.Serial(
        port=port,
        baudrate=baudrate,
        bytesize=serial.EIGHTBITS,
        parity=serial.PARITY_NONE,
        stopbits=serial.STOPBITS_ONE,
        timeout=1
    )
    print("Puerto abierto correctamente. Escanea un código QR ahora...")
    
    while True:
        if ser.in_waiting > 0:
            line = ser.readline()
            print(f"Bytes recibidos: {repr(line)}")
            try:
                decoded = line.decode('utf-8').strip()
                print(f"Decodificado: [{decoded}] (longitud: {len(decoded)})")
            except Exception as e:
                print(f"Fallo al decodificar: {e}")
            sys.stdout.flush()
except KeyboardInterrupt:
    print("\nPrograma terminado por el usuario.")
except Exception as e:
    print(f"Error al abrir o leer el puerto: {e}")
